records/state/state.go

```diff
@@ -103,7 +116,18 @@ func (t *Task) containerIP(src string) string {
 			continue
 		}
 
-		// find the latest docker-inspect label
+		// first try to extract the address from the NetworkInfo structure
+		if len(status.ContainerStatus.NetworkInfos) > 0 {
+			// TODO(CD): handle multiple NetworkInfo objects
+			// TODO(CD): only create A records if the address is IPv4
+			// TODO(CD): create AAAA records if the address is IPv6
+			latestContainerIP = status.ContainerStatus.NetworkInfos[0].IPAddress
```

Comment:
Right, it's not clear to me either how mesos-dns should respond in
situations like that. It also just occurred to me that it might be nice to
include mention of this new behavior somewhere in the mesos-dns docs pages.

On Wed, Sep 16, 2015 at 7:15 PM, Connor Doyle <notifications@github.com>
wrote:

> In records/state/state.go
> <https://github.com/mesosphere/mesos-dns/pull/266#discussion_r39696900>:
>
> > @@ -103,7 +116,18 @@ func (t *Task) containerIP(src string) string {
> >  			continue
> >  		}
> >
> > -		// find the latest docker-inspect label
> > +		// first try to extract the address from the NetworkInfo structure
> > +		if len(status.ContainerStatus.NetworkInfos) > 0 {
> > +			// TODO(CD): handle multiple NetworkInfo objects
> > +			// TODO(CD): only create A records if the address is IPv4
> > +			// TODO(CD): create AAAA records if the address is IPv6
> > +			latestContainerIP = status.ContainerStatus.NetworkInfos[0].IPAddress
>
> I'm not even sure what Mesos-DNS should do when there are multiple
> addreses assigned. I think we'd need conventions for handling additional
> metadata to give us clues (e.g. labels on NetworkInfo, etc.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/266/files#r39696900>.
>